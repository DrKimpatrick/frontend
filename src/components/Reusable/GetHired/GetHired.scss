.main {
  padding-bottom: 0px !important;
}
.get-hired-section {
  .title {
    font-size: 16px;
    width: 100%;
    text-decoration: none solid rgb(116, 116, 116);
    background: linear-gradient(90deg, #3e8074 0%, #1437b1 100%);
    display: flex;
    align-items: center;
    justify-content: center;
    padding: 7px;
  }

  .back-arrow {
    position: absolute;
    color: #747474;
    left: -43px;
    top: 0px;
    font-size: 22px;
  }

  .input-height {
    height: 3rem;
    border-radius: 4px;
    color: #747474;
    font-size: 14px;
  }

  .percent-card {
    width: 71px;
    height: 61px;
    border-radius: 4px;
    background: #4d9b62;
    position: relative;

    span:first-child {
      font-size: 30px;
    }

    span:last-child {
      font-size: 13px;
    }

    .check-icon {
      position: absolute;
      top: -8px;
      right: -8px;
      filter: drop-shadow(0px 1px 1px rgba(0, 0, 0, 0.2));
    }
  }

  .verified-checks {
    background: #ffffff;
    box-shadow: 0px 0px 1px 0px rgba(0, 0, 0, 0.2);
    border: 0.5px solid #ececec;
    border-radius: 4px;

    .check-circle-icon {
      color: #4d9b62 !important;
    }
  }
}

@media (max-width: 786px) {
  .get-hired-section {
    width: 50%;

    .back-arrow {
      display: none;
    }
  }
}

@media (max-width: 500px) {
  .get-hired-section {
    width: 90%;

    .title {
      text-align: center;
    }

    .back-arrow {
      display: none;
    }

    .verified-checks {
      padding-left: 10px;
      padding-right: 10px;
    }
  }
}
