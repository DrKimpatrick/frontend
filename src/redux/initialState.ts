import users from 'redux/initialStates/users';

const initialState = {
  users
};

export default initialState;
