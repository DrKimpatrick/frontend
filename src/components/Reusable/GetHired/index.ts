import GetHired from './GetHired';

export { GetHired };
