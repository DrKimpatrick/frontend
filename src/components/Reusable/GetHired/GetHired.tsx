import React, { FC } from 'react';
import { withRouter, useHistory } from 'react-router-dom';
import ArrowBackTwoToneIcon from '@material-ui/icons/ArrowBackTwoTone';
import ArrowRightAltTwoToneIcon from '@material-ui/icons/ArrowRightAltTwoTone';
import VerifiedUserIcon from '@material-ui/icons/VerifiedUser';
import CheckCircleIcon from '@material-ui/icons/CheckCircle';
import { Routes } from 'utils/routes';
import { MainBackground, NavBar } from 'components/Reusable';
import './GetHired.scss';

const GetHired: FC = () => {
  const history = useHistory();

  return (
    <>
      <NavBar />
      <section className="get-hired-section w-1/3 m-auto text-gray-texts">
        <div className="flex relative h-auto my-8">
          <div className="back-arrow cursor-pointer">
            <ArrowBackTwoToneIcon onClick={() => history.goBack()} />
          </div>
          <div className="font-bold text-xl text-white title">
            <h2>Get Hired</h2>
          </div>
        </div>

        <div className="flex justify-center items-center flex-col mt-16">
          <div className="text-white percent-card flex justify-center items-center ">
            <div>
              <span>86</span>
              <span>%</span>
            </div>
            <VerifiedUserIcon fontSize="large" className="check-icon" />
          </div>
          <div className="w-3/5 text-center mt-5 font-bold">
            Employers are 86% more likely to hire certified and verified talent.
          </div>

          <div className="w-full mt-8 px-12 py-3 verified-checks">
            <div className="flex justify-between items-center my-3">
              <span>Skill Certifications</span>
              <CheckCircleIcon fontSize="small" className="check-circle-icon" />
            </div>
            <div className="flex justify-between items-center my-3">
              <span>Employment Verification</span>
              <CheckCircleIcon fontSize="small" className="check-circle-icon" />
            </div>
            <div className="flex justify-between items-center my-3">
              <span>Education Verification</span>
              <CheckCircleIcon fontSize="small" className="check-circle-icon" />
            </div>
          </div>
        </div>

        <div className="flex justify-center mt-12">
          <button
            data-testid="next-button"
            className="next-btn text-white hover:bg-gray-800 font-semibold py-1 px-3 w-32 rounded-sm shadow flex justify-around"
            type="button"
            onClick={() => history.push(Routes.FeatureChoice)}
          >
            <span className="">Next</span> <ArrowRightAltTwoToneIcon />
          </button>
        </div>
      </section>
      <MainBackground />
    </>
  );
};

export default withRouter(GetHired);
