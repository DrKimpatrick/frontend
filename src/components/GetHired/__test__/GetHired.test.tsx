import { cleanup } from '@testing-library/react';
import React from 'react';
import ReactDom from 'react-dom';
import { BrowserRouter as Router } from 'react-router-dom';
import GetHired from '../GetHired';
import '@testing-library/jest-dom/extend-expect';
import renderer from 'react-test-renderer';
import { Provider } from 'react-redux';
import thunk from 'redux-thunk';
import configureMockStore from 'redux-mock-store';
import initialState from '../../../redux/initialState';

const mockHistoryPush = jest.fn();

let store: any;

const middleware = [thunk];

const mockStore = configureMockStore(middleware);

jest.mock('react-router-dom', () => ({
  ...jest.requireActual('react-router-dom'),
  useHistory: () => ({
    push: mockHistoryPush
  })
}));

describe('`GetHired` component', () => {
  beforeEach(() => {
    store = mockStore(initialState);
  });

  afterEach(cleanup);
  it('renders without crashing', () => {
    const div = document.createElement('div');
    ReactDom.render(
      <Provider store={store}>
        <Router>
          <GetHired />
        </Router>
      </Provider>,
      div
    );
  });

  it('matches the snapshot', () => {
    const tree = renderer
      .create(
        <Provider store={store}>
          <Router>
            <GetHired />
          </Router>
        </Provider>
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
