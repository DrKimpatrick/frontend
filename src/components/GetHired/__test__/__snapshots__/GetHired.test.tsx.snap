// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`\`GetHired\` component matches the snapshot 1`] = `
Array [
  <nav
    className="flex items-center bg-white-800 px-20 py-5 flex-wrap"
  >
    <a
      className="text-gray-800"
      href="/"
      onClick={[Function]}
    >
      <img
        alt="logo"
        className="logo w-24"
        src="logo-image.png"
      />
    </a>
    <div
      className="hidden top-navbar w-full lg:inline-flex lg:flex-grow lg:w-auto"
      id="navigation"
    >
      <div
        className="lg:inline-flex lg:flex-row lg:ml-auto lg:w-auto w-full lg:items-center items-start flex flex-col lg:h-auto"
      >
        <div>
          <a
            className="lg:inline-flex lg:w-auto w-full px-8 py-2 rounded text-gray-600 items-center justify-center hover:text-gray-700"
            href="/"
            onClick={[Function]}
          >
            <span>
              How it works
            </span>
          </a>
          <a
            className="lg:inline-flex lg:w-auto w-full px-8 py-2 rounded text-gray-600 items-center justify-center hover:text-gray-700"
            href="/"
            onClick={[Function]}
          >
            <span>
              About us
            </span>
          </a>
        </div>
        <a
          className="lg:inline-flex lg:w-auto w-full px-8 py-2 rounded text-gray-600 items-center justify-center hover:text-gray-700"
          href="/Login"
          onClick={[Function]}
        >
          <span>
            Login
          </span>
        </a>
        <button
          className="lg:inline-flex lg:w-auto w-full px-8 py-2 rounded items-center justify-center bg-getstartColor hover:bg-blue-800 text-white"
          onClick={[Function]}
        >
          <span>
            Get started
          </span>
        </button>
      </div>
    </div>
  </nav>,
  <section
    className="get-hired-section w-1/3 m-auto text-textGray"
  >
    <div
      className="flex relative h-auto my-8"
    >
      <div
        className="back-arrow cursor-pointer"
      >
        <svg
          aria-hidden={true}
          className="MuiSvgIcon-root"
          focusable="false"
          viewBox="0 0 24 24"
        >
          <path
            d="M20 11H7.83l5.59-5.59L12 4l-8 8 8 8 1.41-1.41L7.83 13H20v-2z"
          />
        </svg>
      </div>
      <div
        className="font-bold text-xl text-white title"
      >
        <h2>
          Get Hired
        </h2>
      </div>
    </div>
    <div
      className="flex justify-center items-center flex-col mt-16"
    >
      <div
        className="text-white percent-card flex justify-center items-center "
      >
        <div>
          <span>
            86
          </span>
          <span>
            %
          </span>
        </div>
        <svg
          aria-hidden={true}
          className="MuiSvgIcon-root check-icon MuiSvgIcon-fontSizeLarge"
          focusable="false"
          viewBox="0 0 24 24"
        >
          <path
            d="M12 1L3 5v6c0 5.55 3.84 10.74 9 12 5.16-1.26 9-6.45 9-12V5l-9-4zm-2 16l-4-4 1.41-1.41L10 14.17l6.59-6.59L18 9l-8 8z"
          />
        </svg>
      </div>
      <div
        className="w-3/5 text-center mt-5 font-bold"
      >
        Employers are 86% more likely to hire certified and verified talent.
      </div>
      <div
        className="w-full mt-8 px-12 py-3 verified-checks"
      >
        <div
          className="flex justify-between items-center my-3"
        >
          <span>
            Skill Certifications
          </span>
          <svg
            aria-hidden={true}
            className="MuiSvgIcon-root check-circle-icon MuiSvgIcon-fontSizeSmall"
            focusable="false"
            viewBox="0 0 24 24"
          >
            <path
              d="M12 2C6.48 2 2 6.48 2 12s4.48 10 10 10 10-4.48 10-10S17.52 2 12 2zm-2 15l-5-5 1.41-1.41L10 14.17l7.59-7.59L19 8l-9 9z"
            />
          </svg>
        </div>
        <div
          className="flex justify-between items-center my-3"
        >
          <span>
            Employment Verification
          </span>
          <svg
            aria-hidden={true}
            className="MuiSvgIcon-root check-circle-icon MuiSvgIcon-fontSizeSmall"
            focusable="false"
            viewBox="0 0 24 24"
          >
            <path
              d="M12 2C6.48 2 2 6.48 2 12s4.48 10 10 10 10-4.48 10-10S17.52 2 12 2zm-2 15l-5-5 1.41-1.41L10 14.17l7.59-7.59L19 8l-9 9z"
            />
          </svg>
        </div>
        <div
          className="flex justify-between items-center my-3"
        >
          <span>
            Education Verification
          </span>
          <svg
            aria-hidden={true}
            className="MuiSvgIcon-root check-circle-icon MuiSvgIcon-fontSizeSmall"
            focusable="false"
            viewBox="0 0 24 24"
          >
            <path
              d="M12 2C6.48 2 2 6.48 2 12s4.48 10 10 10 10-4.48 10-10S17.52 2 12 2zm-2 15l-5-5 1.41-1.41L10 14.17l7.59-7.59L19 8l-9 9z"
            />
          </svg>
        </div>
      </div>
    </div>
    <div
      className="flex justify-center mt-12"
    >
      <button
        className="next-btn text-white hover:bg-gray-800 font-semibold py-1 px-3 w-32 rounded-sm shadow flex justify-around"
        data-testid="next-button"
        onClick={[Function]}
        type="button"
      >
        <span
          className=""
        >
          Next
        </span>
         
        <svg
          aria-hidden={true}
          className="MuiSvgIcon-root"
          focusable="false"
          viewBox="0 0 24 24"
        >
          <path
            d="M16.01 11H4v2h12.01v3L20 12l-3.99-4v3z"
          />
        </svg>
      </button>
    </div>
  </section>,
  <div
    className="mainBackground"
  >
    <img
      alt=""
      src="motif-image.png"
    />
  </div>,
]
`;
